### ISSUE TITLE
Channels are outdated

### ISSUE THREAD

**Issue by creativecreatorormaybenot** [subosito/flutter-action#50]
I think that the action should always provide the **latest** Flutter versions for the specified channels.

For example, the current Dart version in Flutter `stable` is `2.8.4`, however, your action uses Dart `2.7.2`.

**Comment by uziasferreirazup** [comment]
there is any update to this?

**Comment by defuncart** [comment]
I am also experiencing this issue.

```
Running "flutter pub get" in [PROJECT]...                        
The current Flutter SDK version is 1.22.0-12.3.pre.

Because [PROJECT] requires Flutter SDK version >=1.22.0 <2.0.0, version solving failed.
```

The current version of Flutter, however, is 1.22.0 on beta channel (as of 29/09). On local machines, there is no issue in resolving this constraint.

**Comment by subosito** [comment]
I've created a test suite for this, please see https://github.com/subosito/flutter-action-example/runs/1223936570?check_suite_focus=true.

Feel free to reopen if the issue is still persist.

**Comment by defuncart** [comment]
Hi @subosito, thanks for the reply and test.

In your linked test, beta on ubuntu-latest gives the following output

```
Run flutter --version
Flutter 1.22.0-12.3.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision d408d302e2 (8 days ago) • 2020-09-29 11:49:17 -0700
Engine • revision 5babba6c4d
Tools • Dart 2.10.0
```

while stable gives

```
Run flutter --version
Flutter 1.22.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision d408d302e2 (8 days ago) • 2020-09-29 11:49:17 -0700
Engine • revision 5babba6c4d
Tools • Dart 2.10.0
```

However running `flutter --version` on local machine gives

```
flutter --version
Flutter 1.22.0 • channel beta • https://github.com/flutter/flutter.git
Framework • revision d408d302e2 (9 days ago) • 2020-09-29 11:49:17 -0700
Engine • revision 5babba6c4d
Tools • Dart 2.10.0
```

As you can see, although the commit id is the same in all cases, the workflow returns **1.22.0-12.3.pre** as flutter version instead of **1.22.0**.

**Comment by subosito** [comment]
@defuncart using `channel: beta` without specifying `flutter-version` will always fetch the latest one from https://flutter.dev/docs/development/tools/sdk/releases?tab=linux.

Also, I don't see `1.22.0` for beta channel in the release page, not sure why.

**Comment by defuncart** [comment]
@subosito Yes I agree it is strange. 1.22.1 was released yesterday and 1.22.0-12.4.pre for beta, One would expect beta to be ahead of stable 😕  This seems to be a flutter issue. Thanks again for your time and the amazing workflow which I use in all projects :)
